### ISSUE TITLE
[apex] ApexCRUDViolation does not recognise Lists declarated with array syntax

### ISSUE THREAD

**Issue by jh480** [pmd/pmd#4993]
**Affects PMD Version:** 
7.0.0

**Rule:**
https://docs.pmd-code.org/latest/pmd_rules_apex_security.html#ApexCRUDViolationRule

**Description:** Code below only validates list variables declarated with `List<[..]>` , but Apex also allows array Syntax for their List class.

**Code Sample demonstrating the issue:**

```apex
    public void foo() {
        Contact[] contacts = new List<Contact>();

        contacts.add(new Contact());

        if (Contact.sObjectType.getDescribe(SObjectDescribeOptions.DEFERRED).isCreateable()) {
            insert contacts;
        }
    }
```

**Expected outcome:**

PMD reports a violation at line 7, but that's wrong. That's a false positive.

**Running PMD through:** *[Apex PMD (VSC)](https://marketplace.visualstudio.com/items?itemName=chuckjonas.apex-pmd)*
